pkg/cluster/internal/create/actions/loadbalancer/haproxy.go

```diff
@@ -97,7 +98,7 @@ func (a *Action) Execute(ctx *actions.ActionContext) error {
 		"/bin/sh", "-c",
 		fmt.Sprintf(
 			"docker run -d -v /kind/haproxy.cfg:/usr/local/etc/haproxy/haproxy.cfg:ro --network host --restart always --name haproxy %s",
-			haproxy.Image,
+			node.DefaultHAProxyImage,
```

Comment:
if we're going to pre-pull, this needs to come from the node. otherwise this will break between kind versions.

Reply:
do you mean that the node image build has to import the haproxy package to get the image constant; not the other way around?

Reply:
i mean that when we run this container it should match what the image was built with

Reply:
ok i see. so this is a problem..
one possible way of solving that is writing the haproxy image+tag on the node in here:
https://github.com/kubernetes-sigs/kind/blob/978f4a0bc666cc701680eb99da621e1e88448206/pkg/build/node/node.go#L416
(instead of %d)

and then the haproxy container is run we can use the image+tag from there.
but kind of messy.

Reply:
added a different workaround.